src/Libraries/Microsoft.Extensions.AI.Abstractions/CodeInterpreterTool.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Represents a tool that can be specified to an AI service to enable it to execute code it generates.</summary>
+/// <remarks>
+/// This tool does not itself implement code interpration. It is a marker that can be used to inform a service
+/// that the service is allowed to execute its generated code if the service is capable of doing so.
+/// </remarks>
+public class CodeInterpreterTool : AITool
```

Comment:
Do we want to support polymorphic serialization for `AITool` like we do for `AIContent`? I don't believe we have a use case for persisting these objects directly today, but maybe one will come up. 

For `AIFunction` specifically we can't safely roundtrip behavior but it should be possible to roundtrip the schema.

Reply:
It's worth thinking about, but we don't really have a good way to roundtrip the majority case we care about, as you say.